Godeps/_workspace/src/github.com/openshift/source-to-image/pkg/api/types.go

```diff
@@ -91,6 +96,11 @@ type Config struct {
 	// Specify a relative directory inside the application repository that should
 	// be used as a root directory for the application.
 	ContextDir string
+
+	// AllowedUIDs is a list of user ranges of users allowed to run the builder image.
+	// If a range is specified and the builder image uses a non-numeric user or a user
+	// that is outside the specified range, then the build fails.
+	AllowedUIDs user.RangeList
```

Comment:
Yes, I believe this will be sufficient.  If I'm understanding the sequence from the code below it is basically

1.  See if the service account has access to run build pods as root by trying to admit a pod with runAsUser forced to 0
1.  If the above fails set a range of anything except 0 with 1-
1.  The STI builder checks the build image to prevent it from running if the uid is not in the range set by the env var.